crates/matrix-sdk-ui/src/timeline/inner.rs

```diff
@@ -919,20 +921,21 @@ impl<P: RoomDataProvider> TimelineInner<P> {
 
             assert_eq!(state.items.len(), num_items);
```

Comment:
pre-existing, but this check could be hoisted above the `for` loop, maybe?

Reply:
Then it would be entirely pointless, because `num_items` is set to `state.items.len()`. This assert is there to make sure `state.items.len()` doesn't change over multiple iterations of the loop.

Reply:
This is really not obvious from the surrounding code, can you add a comment to that effect?